sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionOwnership.cs

```diff
@@ -0,0 +1,104 @@
+        /// </summary>
+        ///
+        public string PartitionId { get; }
+
+        /// <summary>
+        ///   TODO: purpose not clear yet.
+        /// </summary>
+        ///
+        internal long OwnerLevel { get; }
+
+        /// <summary>
+        ///   The offset of the last <see cref="EventData" /> received by the associated <see cref="IPartitionProcessor" />
+        ///   upon ownership update.
+        /// </summary>
+        ///
+        public long? Offset { get; }
+
+        /// <summary>
+        ///   The sequence number of the last <see cref="EventData" /> received by the associated <see cref="IPartitionProcessor" />
+        ///   upon ownership update.
+        /// </summary>
+        ///
+        public long? SequenceNumber { get; }
+
+        /// <summary>
+        ///   The date and time, in UTC, that the last update was made to this ownership.
+        /// </summary>
+        ///
+        public DateTimeOffset? LastModifiedTime { get; }
+
+        /// <summary>
+        ///   The entity tag needed to update this ownership.
+        /// </summary>
+        ///
+        public string ETag { get; }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="PartitionOwnership"/> class.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The path of the specific Event Hub this partition ownership is associated with, relative to the Event Hubs namespace that contains it.</param>
+        /// <param name="consumerGroup">The name of the consumer group this partition ownership is associated with.</param>
+        /// <param name="instanceId">The identifier of the associated <see cref="EventProcessor" /> instance.</param>
+        /// <param name="partitionId">The identifier of the Event Hub partition this partition ownership is associated with.</param>
+        /// <param name="offset">The offset of the last <see cref="EventData" /> received by the associated <see cref="IPartitionProcessor" />.</param>
+        /// <param name="sequenceNumber">The sequence number of the last <see cref="EventData" /> received by the associated <see cref="IPartitionProcessor" />.</param>
+        /// <param name="lastModifiedTime">The date and time, in UTC, that the last update was made to this ownership.</param>
+        /// <param name="eTag">The entity tag needed to update this ownership.</param>
+        ///
+        public PartitionOwnership(string eventHubName,
+                                  string consumerGroup,
+                                  string instanceId,
+                                  string partitionId,
+                                  long? offset = null,
+                                  long? sequenceNumber = null,
+                                  DateTimeOffset? lastModifiedTime = null,
+                                  string eTag = null)
+        {
+            EventHubName = eventHubName;
```

Comment:
Should we do some validation here to ensure that the important strings aren't null or empty and that values provided for the numerics and dates are non-zero?

Reply:
Just included some checks in `PartitionOwnership` and `Checkpoint`. What do you mean by a "non-zero" date?

Reply:
I read that wrong and was thinking `TimeSpan.Zero`.   But, it may be worth a basic sanity check that it's not `default(DateTimeOffset)`  which is 1/1/0001.    Probably not worth enforcing any specific minimum;  just making sure that it wasn't an unintentionally defaulted value.

Reply:
Per our chat conversation, since we can't figure out a good way to message this, I'm inclined to believe that I'm overthinking.  Let's leave as-is and address if there is an issue.